buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -289,47 +362,130 @@ private void waitForProcessToExit(ProcessHandle processHandle) {
         }
     }
 
-    private File getConfigFile() {
-        return new File(workingDir, "config/elasticsearch.yml");
-    }
-
-    private File getConfPathData() {
-        return new File(workingDir, "data");
-    }
-
-    private File getConfPathSharedData() {
-        return new File(workingDir, "sharedData");
-    }
-
-    private File getConfPathRepo() {
-        return new File(workingDir, "repo");
+    private void createWorkingDir(Path distroExtractDir) {
+        try {
+            syncWithLinks(distroExtractDir, workingDir);
+            Files.createDirectories(confPathRepo);
+            Files.createDirectories(confPathData);
+            Files.createDirectories(workingDir.resolve("sharedData"));
+            Files.createDirectories(confPathLogs);
+            Files.createDirectories(tmpDir);
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed to create working directory: " + workingDir, e);
+        }
     }
 
-    private File getConfPathLogs() {
-        return new File(workingDir, "logs");
+    public void syncWithLinks(Path sourceRoot, Path destinationRoot) throws IOException {
+        // There's some latency in Windows between when a cluster running here previously releases the files so we can
+        // can clean them up. Make sure we can run the same clusters in quick succession.
+        removeWithRetry(destinationRoot);
```

Comment:
This should only happen if we are not waiting on the actual process to die? We should not need to retry deletes...

Reply:
I agree that we shouldn't need this. We do wait for the process to die,
but there's some latency between when we are told that the process is dead and it's resources freed up in Windows. 
I checked that there are no processes left after the build. and the delete can be done manually right after the failure, yet  when this is needed when running the tests. 

It does have some particularity to how the tests are implemented, we run the same project over and over, so the same cluster will be started and stopped over and over.
This is less likely but  not impossible for a real test. 

Without the retry, a test occasionally fails running into this.

Reply:
Sorry I'm still not understanding why we don't have this in our current tests, but it is necessary here. Retries indicate to me something is not waiting correctly.

Reply:
Windows offers no time guarantees on when it removes the locks: 
https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-lockfile
> If a process terminates with a portion of a file locked or closes a file that has 
>outstanding locks, the locks are unlocked by the operating system. However, the 
>time it takes for the operating system to unlock these locks depends upon 
>available system resources. Therefore, it is recommended that your process 
>explicitly unlock all files it has locked when it terminates. If this is not done, access 
>to these files may be denied if the operating system has not yet unlocked them.

As to why this doesn't show up in the existing tests, the explanation is that we don't have tests for `ClusterFormationTasks` and the tests that use them never re-spin the same cluster, even  the upgrade tasks set up a new node with an old data path. 
The integration tests for testclusters run a build that starts and stops a cluster in quick succession leading to the same cluster starting and stopping multiple times for different Gradle task graphs. This led to failures on Windows only and occasionally. 
The retry is in a way in place to make the test pass, but technically it's a possible use-case for test clusters even if unlikely.